File: OnlineSurveyTool.Server.Services/DTOs/UserRegisterDTO.cs
using System.ComponentModel.DataAnnotations;

namespace OnlineSurveyTool.Server.Services.DTOs
{
    public class UserRegisterDTO
    {
        [Required]
        [StringLength(32, MinimumLength = 8)]
        public string Login { get; set; }
        [Required]  
        [StringLength(100)]
        public string EMail { get; set; }
        [Required]
        [StringLength(64, MinimumLength = 8)]
        public string Password { get; set; }
    }
}

File: onlinesurveytool.client/src/app/models/user.register.dto.ts
export class UserRegisterDTO {
  login: string | null | undefined;
  email: string | null | undefined;
  password: string | null | undefined;

  constructor(login: string | null | undefined, email: string | null | undefined, password: string | null | undefined) {
    this.login = login;
    this.email = email;
    this.password = password;
  }
}
